salt/modules/mount.py

```diff
@@ -893,26 +901,26 @@ def set_automaster(
                         comps[2] = device_fmt
                     if change:
                         log.debug(
-                            'auto_master entry for mount point {0} needs to be '
-                            'updated'.format(name)
-                       )
+                            'auto_master entry for mount point %s needs to be '
+                            'updated', name
+                        )
                         newline = (
                             '{0}\t{1}\t{2}\n'.format(
                                 name, type_opts, device_fmt)
-                       )
+                        )
                         lines.append(newline)
                 else:
                     lines.append(line)
     except (IOError, OSError) as exc:
         msg = 'Couldn\'t read from {0}: {1}'
-        raise CommandExecutionError(msg.format(config, str(exc)))
+        raise CommandExecutionError(msg.format(config, six.text_type(exc)))
 
     if change:
         if not salt.utils.args.test_mode(test=test, **kwargs):
             try:
                 with salt.utils.files.fopen(config, 'w+') as ofile:
                     # The line was changed, commit it!
-                    ofile.writelines(lines)
+                    ofile.writelines(salt.utils.data.decode(lines))
```

Comment:
Same as above, this needs to be an `encode`, and we need to open the file with a mode of `'wb'`.